src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -316,6 +316,15 @@ public void updateCertificateObjects(SecuredUUID uuid, CertificateUpdateObjectsD
         if (request.getOwnerUuid() != null) {
             updateOwner(uuid, request.getOwnerUuid());
         }
+
+        if (request.getTrustedCa() != null) {
+            updateTrustedCaMark(uuid, request.getTrustedCa());
+        }
+    }
+
+    private void updateTrustedCaMark(SecuredUUID uuid, Boolean trustedCa) throws NotFoundException {
+        Certificate certificate = getCertificateEntity(uuid);
+        if (certificate.getTrustedCa() != null) certificate.setTrustedCa(trustedCa);
```

Comment:
Throw here ValidationException when trying to set certificate as trusted and it is not CA certificate. It will need to add it to throws clause in Interfaces.